File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3106_lexicographically-smallest-string-after-operations-with-constraint/solutions/java/solution.java
import java.util.Arrays;

class Solution {
    public String getSmallestString(String s, int k) {
        char[] t = s.toCharArray();
        int n = t.length;

        for (int i = 0; i < n; i++) {
            if (k <= 0) {
                break;
            }
            char current = t[i];
            // Find the minimum possible character we can change to while adjusting the distance
            char bestChar = 'a';
            int bestDistance = Math.min(Math.abs(current - 'a'), 26 - Math.abs(current - 'a'));

            if (bestDistance <= k) {
                t[i] = bestChar;
                k -= bestDistance;
            }
        }

        return new String(t);
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3106_lexicographically-smallest-string-after-operations-with-constraint/solutions/python3/solution.py
class Solution:
    def getSmallestString(self, s: str, k: int) -> str:
        def cyclic_distance(c1, c2):
            return min(abs(ord(c1) - ord(c2)), 26 - abs(ord(c1) - ord(c2)))
        
        n = len(s)
        t = list(s)
        
        for i in range(n):
            # Try to convert s[i] to 'a' if possible within the remaining cost k
            if k > 0:
                # Calculate minimum distance to 'a'
                dist = cyclic_distance(s[i], 'a')
                if dist <= k:
                    t[i] = 'a'
                    k -= dist
        
        return ''.join(t)
